### ISSUE TITLE
pycbc_generate_hwinj is broken

### ISSUE THREAD

**Issue by spxiwh** [ligo-cbc/pycbc#1338]
The lines:

```
if not opts.psd_estimation and (opts.frame_files or opts.frame_type
                            or opts.frame_cache or opts.fake_strain):
    raise KeyError("Must use --psd-estimation with frame options"
                   "(--frame-files, --frame-type, --frame-cache, "
                   "and --fake-strain).")
```

have broken pycbc_generate_hwinj, because these always evaluate to True. Reason: These are all multi-detector defaultdict types, which evaluates to True even when empty. Probably the best thing to do is for me to sub-class the defaultdict's __bool__ attribute (or whatever it is in python2.X) to evaluate to False if empty.

**Comment by spxiwh** [comment]
This seems broken in a few other places as well.

**Comment by cmbiwer** [comment]
>>  Probably the best thing to do is for me to sub-class the defaultdict's bool attribute (or whatever it is in python2.X) to evaluate to False if empty.

I like this suggestion and then I don't think anything in ``pycbc_generate_hwinj`` needs to be changed.

**Comment by spxiwh** [comment]
There were two other issues with this code:
```
358,362d357
<     # get time delay to detector from center of the Earth
<     time_delay = det.time_delay_from_earth_center(sim.longitude, sim.latitude,
<                                                   sim.geocent_end_time)
<     end_time = sim.geocent_end_time + time_delay
< 
418c413
< for ifo in opts.instruments:
---
> for ifo in opt.instruments:
```

... Posting this here so I don't lose track when I get a chance to look at the MultiIfo option stuff

**Comment by cmbiwer** [comment]
Ian fixed this in #1342 so I'll close. Thanks Ian.
